pkg/agent/proxy/proxier.go

```diff
@@ -189,21 +235,44 @@ func (p *proxier) installServices() {
 			p.endpointInstalledMap[svcPortName] = nil
 			continue
 		}
+		// Delete previous flow.
+		if needRemoval {
+			if err := p.ofClient.UninstallServiceFlows(pSvcInfo.ClusterIP(), uint16(pSvcInfo.Port()), pSvcInfo.OFProtocol); err != nil {
+				klog.Errorf("Failed to remove flows of Service %v: %v", svcPortName, err)
+			}
+		}
 		if err := p.ofClient.InstallServiceFlows(groupID, svcInfo.ClusterIP(), uint16(svcInfo.Port()), svcInfo.OFProtocol, uint16(svcInfo.StickyMaxAgeSeconds())); err != nil {
 			klog.Errorf("Error when installing Service flows: %v", err)
 			continue
 		}
 		// Install OpenFlow entries for the ingress IPs of LoadBalancer Service.
-		// The LoadBalancer Service should can be accessed from Pod, Node and
+		// The LoadBalancer Service should be accessible from Pod, Node and
 		// external host.
-		for _, ingress := range svcInfo.LoadBalancerIPStrings() {
+		var toDelete, toAdd []string
+		if needRemoval {
+			toDelete = pSvcInfo.LoadBalancerIPStrings()
+			toAdd = svcInfo.LoadBalancerIPStrings()
+		} else {
+			toDelete = deletedLoadBalancerIPs
+			toAdd = addedLoadBalancerIPs
+		}
+		for _, ingress := range toDelete {
+			if ingress != "" {
+				if err := p.uninstallLoadBalancerServiceFlows(net.ParseIP(ingress), uint16(svcInfo.Port()), svcInfo.OFProtocol); err != nil {
```

Comment:
should it use pSvcInfo.Port() and pSvcInfo.OFProtocol?

Reply:
Fixed, thanks